app/domain/errors.rb

```diff
@@ -0,0 +1,213 @@
+          code: "CONJ00012E"
+        )
+
+        IdTokenFieldNotFoundOrEmpty = ::Util::ErrorClassWithCode.new(
+          msg: "Field '{0}' not found or empty in ID Token",
+          code: "CONJ00013E"
+        )
+
+        IdTokenInvalidFormat = ::Util::ErrorClassWithCode.new(
+          msg: "Invalid ID Token Format (3rdPartyError ='{0}')",
+          code: "CONJ00014E"
+        )
+
+        IdTokenVerifyFailed = ::Util::ErrorClassWithCode.new(
+          msg: "ID Token verification failed (3rdPartyError ='{0}')",
+          code: "CONJ00015E"
+        )
+
+        IdTokenExpired = ::Util::ErrorClassWithCode.new(
+          msg: "ID Token Expired",
+          code: "CONJ00016E"
+        )
+
+        AdminAuthenticationDenied = ::Util::ErrorClassWithCode.new(
+          msg: "admin user is not allowed to authenticate with OIDC",
+          code: "CONJ00017E"
+        )
+
+      end
+
+      module AuthnIam
+
+        InvalidAWSHeaders = ::Util::ErrorClassWithCode.new(
+          msg: "'Invalid or Expired AWS Headers: {0}",
+          code: "CONJ00018E"
+        )
+
+      end
+
+      module AuthnK8s
+
+        WebserviceNotFound = ::Util::ErrorClassWithCode.new(
+          msg: "Webservice '{0}' wasn't found",
+          code: "CONJ00019E"
+        )
+
+        HostNotFound = ::Util::ErrorClassWithCode.new(
+          msg: "Host '{0}' wasn't found",
+          code: "CONJ00020E"
+        )
+
+        HostNotAuthorized = ::Util::ErrorClassWithCode.new(
+          msg: "'{0}' does not have 'authenticate' privilege on {1}",
+          code: "CONJ00021E"
+        )
+
+        CSRIsMissingSpiffeId = ::Util::ErrorClassWithCode.new(
+          msg: 'CSR must contain SPIFFE ID SAN',
+          code: "CONJ00022E"
+        )
+
+        CSRNamespaceMismatch = ::Util::ErrorClassWithCode.new(
+          msg: "Namespace in SPIFFE ID '{0}' must match namespace implied by common name '{1}'",
+          code: "CONJ00023E"
+        )
+
+        PodNotFound = ::Util::ErrorClassWithCode.new(
+          msg: "No Pod found for podname '{0}' in namespace '{1}'",
+          code: "CONJ00024E"
+        )
+
+        ScopeNotSupported = ::Util::ErrorClassWithCode.new(
+          msg: "Resource type '{0}' identity scope is not supported in this version " +
+            "of authn-k8s",
+          code: "CONJ00025E"
+        )
+
+        ControllerNotFound = ::Util::ErrorClassWithCode.new(
+          msg: "Kubernetes {0} {1} not found in namespace {2}",
+          code: "CONJ00026E"
+        )
+
+        CertInstallationError = ::Util::ErrorClassWithCode.new(
+          msg: "Cert could not be copied to pod: {0}",
+          code: "CONJ00027E"
+        )
+
+        ContainerNotFound = ::Util::ErrorClassWithCode.new(
+          msg: "Container {0} was not found for requesting pod",
+          code: "CONJ00028E"
+        )
+
+        MissingClientCertificate = ::Util::ErrorClassWithCode.new(
+          msg: "The client SSL cert is missing from the header",
```

Comment:
Client SSL certificate...
(no need for "The" because we haven't used it anywhere else